scripts/garmin_sync.py

```diff
@@ -341,64 +200,75 @@ def __init__(self, status):
         self.status = status
 
 
-def download_garmin_gpx(client, activity_id):
+async def download_garmin_gpx(client, activity_id):
     try:
-        gpx_data = client.download_activity(
-            activity_id, dl_fmt=client.ActivityDownloadFormat.GPX
-        )
-        file_path = os.path.join(GPX_FOLDER, activity_id + ".gpx")
-        with open(file_path, "wb") as fb:
-            fb.write(gpx_data)
+        gpx_data = await client.download_activity(activity_id)
+        file_path = os.path.join(GPX_FOLDER, f"{activity_id}.gpx")
+        async with aiofiles.open(file_path, "wb") as fb:
+            await fb.write(gpx_data)
     except:
-        print(f"wrong id {activity_id}")
+        print(f"Failed to download activity {activity_id}: ")
+        traceback.print_exc()
         pass
 
 
+async def get_activity_id_list(client, start=0):
+    activities = await client.get_activities(start, 100)
+    if len(activities) > 0:
+        ids = list(map(lambda a: a.get("activityId"), activities))
+        logger.debug(f"Activity IDs: {ids}")
+        return ids + await get_activity_id_list(client, start + 100)
```

Comment:
cool, I use it in [runtastic](https://github.com/yihong0618/Runtastic/blob/master/runtastic.py#L259) too~